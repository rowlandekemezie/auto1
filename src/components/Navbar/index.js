import React from "react";
import { NavLink } from "react-router-dom";
import styled from "styled-components";

import logo from "../../static/auto1_logo@2x.png";

const Wrapper = styled("div")`
  display: flex;
  max-height: 80px;
  padding: 12px;

  @media (min-width: 768px) {
    padding: 24px;
  }
`;

const NavItemsWrapper = styled("div")`
  display: flex;
  margin-left: auto;
  align-items: center;
  justify-content: space-between;
`;

const BrandLogo = styled(NavLink)`
  width: 150px;
  height: 32px;
`;

const Logo = styled("img")`
  width: 100%;
  height: 100%;
`;

const StyledNavLink = styled(NavLink)`
  text-decoration: none;
  color: ${props => props.theme.BLACK};
  font-size: 14px;
  margin-left: 20px;
`;

const renderLink = (page, i) => (
  <StyledNavLink
    key={i}
    to={`/`} /** Should be a real page when this feature is provided **/
  >
    {page}
  </StyledNavLink>
);

const Navbar = () => {
  const pages = ["Purchase", "My Orders", "Sell"];

  return (
    <Wrapper>
      <BrandLogo to={"/"}>
        <Logo src={logo} />
      </BrandLogo>
      <NavItemsWrapper>{pages.map(renderLink)}</NavItemsWrapper>
    </Wrapper>
  );
};

export default Navbar;
