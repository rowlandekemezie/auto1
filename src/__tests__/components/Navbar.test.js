import React from "react";
import { shallow, mount } from "enzyme";
import { Router } from "react-router-dom";
import history from "history/createBrowserHistory";
import Navbar from "../../components/Navbar";

describe("<Navbar />", () => {
  it("should render component properly", () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper.length).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correct <img /> and <Link /> tags", () => {
    const renderComp = mount(
      <Router history={history()}>
        <Navbar />
      </Router>
    );
    expect(renderComp.find("img").length).toBe(1);
    expect(renderComp.find("Link").length).toBe(4);
  });
});
