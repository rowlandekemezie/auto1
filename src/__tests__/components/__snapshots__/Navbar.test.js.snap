// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render component properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          to="/"
        >
          <ForwardRef
            src="auto1_logo@2x.png"
          />
        </ForwardRef>,
        <ForwardRef>
          <ForwardRef
            to="/"
          >
            Purchase
          </ForwardRef>
          <ForwardRef
            to="/"
          >
            My Orders
          </ForwardRef>
          <ForwardRef
            to="/"
          >
            Sell
          </ForwardRef>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            src="auto1_logo@2x.png"
          />,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "src": "auto1_logo@2x.png",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "rules": Array [
                "
  width: 100%;
  height: 100%;
",
              ],
            },
            "displayName": "styled.img",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "img",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  width: 150px;
  height: 32px;
",
            ],
          },
          "displayName": "Styled(NavLink)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef
              to="/"
            >
              Purchase
            </ForwardRef>,
            <ForwardRef
              to="/"
            >
              My Orders
            </ForwardRef>,
            <ForwardRef
              to="/"
            >
              Sell
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "children": "Purchase",
              "to": "/",
            },
            "ref": null,
            "rendered": "Purchase",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": false,
                "rules": Array [
                  "
  text-decoration: none;
  color: ",
                  [Function],
                  ";
  font-size: 14px;
  margin-left: 20px;
",
                ],
              },
              "displayName": "Styled(NavLink)",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "children": "My Orders",
              "to": "/",
            },
            "ref": null,
            "rendered": "My Orders",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": false,
                "rules": Array [
                  "
  text-decoration: none;
  color: ",
                  [Function],
                  ";
  font-size: 14px;
  margin-left: 20px;
",
                ],
              },
              "displayName": "Styled(NavLink)",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "children": "Sell",
              "to": "/",
            },
            "ref": null,
            "rendered": "Sell",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": false,
                "rules": Array [
                  "
  text-decoration: none;
  color: ",
                  [Function],
                  ";
  font-size: 14px;
  margin-left: 20px;
",
                ],
              },
              "displayName": "Styled(NavLink)",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  margin-left: auto;
  align-items: center;
  justify-content: space-between;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  max-height: 80px;
  padding: 12px;

  @media (min-width: 768px) {
    padding: 24px;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            to="/"
          >
            <ForwardRef
              src="auto1_logo@2x.png"
            />
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef
              to="/"
            >
              Purchase
            </ForwardRef>
            <ForwardRef
              to="/"
            >
              My Orders
            </ForwardRef>
            <ForwardRef
              to="/"
            >
              Sell
            </ForwardRef>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              src="auto1_logo@2x.png"
            />,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "src": "auto1_logo@2x.png",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
  width: 100%;
  height: 100%;
",
                ],
              },
              "displayName": "styled.img",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "img",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  width: 150px;
  height: 32px;
",
              ],
            },
            "displayName": "Styled(NavLink)",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                to="/"
              >
                Purchase
              </ForwardRef>,
              <ForwardRef
                to="/"
              >
                My Orders
              </ForwardRef>,
              <ForwardRef
                to="/"
              >
                Sell
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "children": "Purchase",
                "to": "/",
              },
              "ref": null,
              "rendered": "Purchase",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "rules": Array [
                    "
  text-decoration: none;
  color: ",
                    [Function],
                    ";
  font-size: 14px;
  margin-left: 20px;
",
                  ],
                },
                "displayName": "Styled(NavLink)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "children": "My Orders",
                "to": "/",
              },
              "ref": null,
              "rendered": "My Orders",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "rules": Array [
                    "
  text-decoration: none;
  color: ",
                    [Function],
                    ";
  font-size: 14px;
  margin-left: 20px;
",
                  ],
                },
                "displayName": "Styled(NavLink)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "children": "Sell",
                "to": "/",
              },
              "ref": null,
              "rendered": "Sell",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "rules": Array [
                    "
  text-decoration: none;
  color: ",
                    [Function],
                    ";
  font-size: 14px;
  margin-left: 20px;
",
                  ],
                },
                "displayName": "Styled(NavLink)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  margin-left: auto;
  align-items: center;
  justify-content: space-between;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  max-height: 80px;
  padding: 12px;

  @media (min-width: 768px) {
    padding: 24px;
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
